### ISSUE TITLE
Discussion about beta * kl loss term exploding.

### ISSUE THREAD

**Issue by curranw** [woonsangcho/trpo#1]
I heavily based my code on both your code and Dr. Pat Coady (https://github.com/pat-coady/trpo).

In both code bases, after a while PPO overfits and unlearns everything within a couple of updates. I'm assuming it's overfitting, because the beta * kl term becomes large after the policy has converged to a decent policy. Once this happens, the policy is completely destroyed by a large loss function. For example, early in a run the losses look like:
loss1: -0.0036041432 loss2: 0.010586119 loss3: 0.000427304

where loss1 is the advantage loss, loss2 is beta * kl and loss3 is the hinge loss. These make sense! 

After convergence, the losses get a little crazier:
loss1: -0.024061535 loss2: 8.501843 loss3: 17.157572

this is due to kl exploding after a single update, and beta getting consistently larger (until 20). I tried to fix this by trying:

1. L2 regularization on the policy, but it only delays the effect. 
2. Small minibatch updates per actor epoch, rather than one big update. This helps a lot, but it still occurs eventually.

Any thoughts?

Note that this only occurs when updating the actor/critic > 3 times per update.
